lucene/core/src/java/org/apache/lucene/index/BaseCompositeReader.java

```diff
@@ -89,7 +88,6 @@ protected BaseCompositeReader(R[] subReaders) throws IOException {
 
     this.maxDoc = Math.toIntExact(maxDoc);
     starts[subReaders.length] = this.maxDoc;
-    this.numDocs = Math.toIntExact(numDocs);
```

Comment:
I guess it was done because we didn't always check the maximum number of documents allowed in an index but now we have better checks so we can safely assume that the sum will not overflow ?

Reply:
I think it's not necessary on numDocs. By the time we compute numDocs(), we already checked that maxDoc values summed up to a number that doesn't overflow. Since numDocs is less than or equal to maxDoc, the sum of numDocs shouldn't overflow either?